swarm/src/handler/one_shot.rs

```diff
@@ -50,6 +49,26 @@ where
     config: OneShotHandlerConfig,
 }
 
+impl<TInbound, TOutbound, TEvent> Debug for OneShotHandler<TInbound, TOutbound, TEvent>
+where
+    TInbound: Debug,
+    TOutbound: OutboundUpgradeSend + Debug,
+    TOutbound::Error: Debug,
+    TEvent: Debug,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("OneShotHandler")
+            .field("listen_protocol", &self.listen_protocol)
+            .field("pending_error", &self.pending_error)
+            .field("events_out", &self.events_out)
+            .field("dial_queue", &self.dial_queue)
+            .field("dial_negotiated", &self.dial_negotiated)
+            .field("keep_alive", &self.keep_alive)
+            .field("config", &self.config)
+            .finish()
+    }
+}
```

Comment:
Same here, you are not excluding any fields. Why the manual implementation?

Reply:
I am not sure we should implement `Debug` on things like `ConnectionHandler`s and `NetworkBehaviour` structs. Those often contain data that cannot be printed easily, so unless we can `derive` it, I'd vote against adding this.

Reply:
Because of the where clause exist only on the `impl Debug` and not on the struct, the user is not restricted to have the Debug trait implemented for their types.

Reply:
Do you feel strongly about adding debug to this struct? I think it is rather odd to debug-print the entire struct. As a user, you never "own" one of these, instead you only have access to it in methods via `self`. Wouldn't it make more sense to just debug-print the fields in question?

I'd be open to adding it if we could derive it but a manual implementation seems to be too verbose for what I think is a niche usecase.

Reply:
No strong opinion for this one. I can remove it.

Reply:
removed